diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index d234c0e..2681c57 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -19,6 +19,8 @@

 file(GLOB PULSAR_SOURCES *.cc *.h lz4/*.cc lz4/*.h checksum/*.cc checksum/*.h stats/*.cc stats/*.h c/*.cc c/*.h auth/*.cc auth/*.h auth/athenz/*.cc auth/athenz/*.h)

+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
 if (NOT PROTOC_PATH)
     set(PROTOC_PATH protoc)
 endif()
